packages/react-server/core/ReactServerAgent/Request.js

```diff
@@ -233,7 +228,35 @@ function buildSuperagentRequest() {
 		}
 	}
 
-	req.send(postParams);
+	let hasQueryParams = (this._queryParams.length > 0),
+		hasPostParams = (Object.keys(this._postParams).length > 0);
+
+	if (hasPostParams) {
+		// superagent has some weird, implicit file upload support
+		// that only works if you don't set `type`.
+		if (this._type && this._type !== 'form-data') {
+			req.type(this._type);
+		}
+
+		req.send(postParams);
+	}
```

Comment:
Correct...any thoughts on the right way to handle this?  Here's the basic situation: on GET/HEAD requests, if we set `type` and/or `send` to any value (even empty values), the GET/HEAD requests will send a CORS preflight.